packages/grafana-data/src/transformations/transformers/joinDataFrames.ts

```diff
@@ -362,6 +362,77 @@ function joinOuterTabular(
   return data;
 }
 
+/**
+ * This function performs a sql-style inner join on tabular data;
+ * it will combine records from two tables whenever there are matching
+ * values in a field common to both tables.
+ *
+ * NOTE: This function implicitly assumes that the first array in each AlignedData
+ * contains the values to join on. It doesn't explicitly specify a column field to join on,
+ * but rather uses the 0th position of the arrays (AlignedData[0]) to determine the joining keys.
+ * Then, when processing the tables, the function iterates over the values in the `xValues`
+ * (the joining keys) array and checks if the current row `currentRow` already includes the value.
+ * If a matching value is found, it joins the corresponding values from the remaining arrays `yValues`
+ * (all other non-joining key arrays) to create a new row in the joined table.
+ *
+ * @param {AlignedData[]} tables - The tables to join.
+ *
+ * @returns {Array<Array<string | number | null | undefined>>} The joined tables as an array of arrays, where each array represents a row in the joined table.
+ */
+function joinInner(tables: AlignedData[]): Array<Array<string | number | null | undefined>> {
+  const joinedTables: Array<Array<string | number | null | undefined>> = [];
+
+  // Helper function to get the value at a specific index from an array or typed array
+  const getValue = (arr: number[] | TypedArray | Array<number | null | undefined>, index: number) => arr[index];
```

Comment:
dont need this

Reply:
Don't need what?

Reply:
You mean I could remove it now that I'm not accessing via the `at` method? Yeah, that's probably a good idea.

Reply:
Removed.